### PR TITLE
Clean up diagnosticserver socket on unexpected shutdown

### LINKED ISSUE #25968: dotnet-diagnostic sockets lingering after crash
.NET Core version: 3.0.0-preview7-27912-14 Linux

My target is an arm64 embedded platform, but I am seeing the same behaviour on my x64 dev machine.

The runtime is leaving these diagnostic sockets with names like `dotnet-diagnostic-28148-23379390-socket` under /tmp if the process exits with an unhandled exception. If the process is repeatedly crashing in this way then these sockets pile up, using more and more memory (/tmp is mounted in RAM on my system), eventually overwhelming the system. Is there a way to prevent these sockets from being created, or ensuring that they are cleaned up properly on an unhandled exception?

How I am running my executable:
```
$ dotnet test.dll
Unhandled exception. [Trim]
Aborted
```

### PR DESCRIPTION
**Pull request by sywhang**
Fixes #25968

### CODE REVIEW

File: src/debug/ee/debugger.cpp [hunk 1]
```diff
@@ -898,6 +899,7 @@ AbortTransport()
     {
         g_pDbgTransport->AbortConnection();
     }
+    DiagnosticServer::Shutdown();
```
Comment (noahfalk) [inline src/debug/ee/debugger.cpp]: I'm skeptical this is the best place to insert this call. Conceptually the diagnostics server has no dependency on the debugger, they are two separate components that both happen to use a pipe. I would look up the callstack for the error handling code that is outside the debugger and hook the diagnostics server shutdown code there.

Comment (sywhang) [reply src/debug/ee/debugger.cpp]: Thanks for the feedback Noah! I just updated the PR with your suggestion :)

File: src/debug/ee/debugger.cpp [hunk 2]
```diff
@@ -2047,9 +2059,6 @@ HRESULT Debugger::Startup(void)
             ShutdownTransport();
             ThrowHR(hr);
         }
-    #ifdef FEATURE_PAL
-        PAL_SetShutdownCallback(AbortTransport);
```
Comment (noahfalk) [inline src/debug/ee/debugger.cpp]: AbortTransport is now dead code, so it should also be removed.

**Comment by MeiChin-Tsai** [comment]
Can you tell me what are being tested for this fix?

**Comment by sywhang** [comment]
@MeiChin-Tsai I tested via manual testing on Ubuntu x64 and macOS.

**Pull request merged**